app/src/modules/settings/routes/data-model/field-detail/field-detail-simple/field-detail-simple.vue

```diff
@@ -57,7 +57,7 @@ export default defineComponent({
 		},
 		search: {
 			type: String,
-			required: true,
+			default: null,
```

Comment:
Wondering what's the cleaner solution to get rid of the current type mismatch in `app/src/modules/settings/routes/data-model/field-detail/field-detail.vue` 😄 

- Changing it to 
  ```suggestion
				type: String as () => string | null,
				required: true,
  ```
- or adjusting the line here https://github.com/directus/directus/blob/678c5f57af547f567b8c12f0c67e471f82d70710/app/src/modules/settings/routes/data-model/field-detail/field-detail.vue#L78 to 
  ```js
 		const search = ref<string | undefined>(undefined); 
  ```

Reply:
I've never seen the `String as () => string | null` syntax before. It doesn't seem to work when trying it out locally. And as far as I can see it's not supported by Vue? https://github.com/vuejs/vue/issues/1961

I know it works for sure when using a `<script setup>` block and `search: string | null`. So not really sure what to do here.

Reply:
Oh, right this just makes TypeScript happy but not Vue 😅 

An ugly workaround for this would be https://github.com/vuejs/core/issues/3948#issuecomment-860466204.

Would you mind adjusting `field-detail.vue` as suggested above in that case? Thx!

Reply:
hey psst would it be fixed if we rework it to a `<script setup>` with `defineProps`? I wanna move everything over regardless, so this could be a good excuse

Reply:
Agreed. I'll use that excuse

Reply:
That's one ugly diff. Looks wait better with "Hide Whitespace"

Reply:
Awesome 🚀